src/wallet/rpc/transactions.cpp

```diff
@@ -137,6 +137,8 @@ static UniValue ListReceived(const CWallet& wallet, const UniValue& params, cons
     const auto& func = [&](const CTxDestination& address, const std::string& label, const std::string& purpose, bool is_change) {
         if (is_change) return; // no change addresses
 
+        if (purpose == "send") return; // no send addresses
```

Comment:
In commit "wallet: Filter-out "send" addresses from `listreceivedby*`" (ce7fc736576664701b09bea811d77113948c38bb)

It like it would be better to call IsMine here instead of relying on the purpose field. The PR description says this fix only covers the include_empty=true, include_watchonly=true case, not the include_watchonly=false case, so maybe you could fix both cases by using the `isminetype` result returned by `IsMine` instead of purpose == "send".

Also, I'm not 100% sure about this, and am curious about other opinions, but in general the purpose field seems like it is set pretty haphazardly in code, so things that rely on it could easily break, and probably better not to use it when alternatives are available.